Hub/Services/ActivityTemplate.cs

```diff
@@ -378,8 +378,11 @@ public void RegisterOrUpdate(ActivityTemplateDO activityTemplateDo)
                     // We're creating new ActivityTemplate.
                     if (activity == null)
```

Comment:
If activity is found we should check if found activity has the same Id as the activity that was passed. If ids are different we should throw exception. It should no be possible to change Id of already registered activity template.
So in the beginning of the 'else' block add the following code:

if (activity.Id != activityTemplateDo.Id)
{
	throw new InvalidOperationException ("Existent activity with same Name and Version that we passed " 
	+ "has different Id. Changing of activity template Id is not possible. If you need to have another Id please update the version number or create new activity template")
}